modules/core/src/utils/flatten.js

```diff
@@ -142,3 +142,85 @@ export function flattenTypedVertices(nestedArray, {
   return start;
 }
 */
+
+/*
+ * Helper function for padArray
+ */
+function padSubArray({source, target, start = 0, end, enter}) {
+  end = end || target.length;
+
+  const sourceLength = source.length;
+  const targetLength = end - start;
+
+  if (sourceLength > targetLength) {
+    target.set(source.subarray(0, targetLength), start);
+    return;
+  }
+
+  target.set(source, start);
+
+  // source is not large enough to fill target space, call `enter` to get filler data
+  let i = sourceLength;
+  while (i < targetLength) {
+    const datum = enter(i, source);
+    for (let j = 0; j < datum.length; j++) {
+      target[start + i] = datum[j];
+      i++;
+    }
+  }
+}
+
+/*
+ * Use the source data array to fill the target data array
+ * @params {TypedArray} source - original data
+ * @params {TypedArray} target - output data
+ * @params {Number} size - length per datum
+ * @params {Function} enter - callback to get new data when source is short
```

Comment:
`enter` is a rather odd name? 

`getData`? 

Could it default to `() => 0`, or it pretty much always needs to be supplied?